### ISSUE TITLE
Arguments to FannieTask->run()

### ISSUE THREAD

**Issue by flathat** [CORE-POS/IS4C#458]
I have a lane that I sometimes use for tests.  I'd like to be able to sync it without syncing the other lanes.

I use the cronjob `$CRON/lanesync.api.php`, which I've made into Fannie Task `LaneSyncApiTask.php` (not published yet) and want to run it from the command line, with a lane# argument, to sync the one lane:
`$ sudo -u www-data php $FCL2/FannieTask.php LaneSyncApiTask 3`

I've done this by adding at the end of the `(php_sapi_name == 'cli' ...)` branch of `FannieTask.php`
```php
    if ($argc > 2) {
        $params = array_slice($argv,2);
        $obj->run($params);
    } else {
        $obj->run();
    }
```
instead of just
```php
$obj->run();
```
and in the Task:
```php
    public function run($lane=array(0))
    {
        global $FANNIE_LANES, $FANNIE_COMPOSE_LONG_PRODUCT_DESCRIPTION,
                $FANNIE_COOP_ID;

        $oneLane = $lane[0];
        if ($oneLane > count($FANNIE_LANES)) {
            echo $this->cronMsg("oneLane: $oneLane is more than the ".
                count($FANNIE_LANES)." that are configured.");
            return;
        }
```
and code in `SyncLanes->pushTable()` (and similar) to handle the additional argument:
```php
            $result = SyncLanes::pushTable("$table", 'op',
                SyncLanes::TRUNCATE_DESTINATION,
                $oneLane);
```
Are there risks or problems with this I haven't foreseen?  Better ways of doing it?

(I wanted to pass the arguments to `run()` as a list: `run(p1[,p2 ...])` instead of an array but couldn't figure out how to do it. Also possibly make them all strings.)

**Comment by gohanman** [comment]
Changing the method signature of run() means either that class will raise NOTICEs or every FannieTask needs to be updated (if the signature does change on every class it should be something more generic like $argv=array()).

You could pass arguments sequentially using call_user_func_array to call run() and then using func_get_arg/func_num_args to read them in run().

I think the better way is something like this: 8cd65a04107aaa08d13c5196530419198b30af05. Might as well allow for named options so complex CLI commands can be more structured.

**Comment by flathat** [comment]
I can report success using https://github.com/CORE-POS/IS4C/commit/8cd65a04107aaa08d13c5196530419198b30af05 from the command line with an argument. I haven't yet used it for regular Automated Tasks.

**Comment by gohanman** [comment]
Great! Included in 1.2.0
